src/librustc_trans/back/symbol_names.rs

```diff
@@ -162,11 +173,34 @@ fn get_symbol_hash<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     });
 
     // 64 bits should be enough to avoid collisions.
-    format!("h{:016x}", hasher.finish())
+    hasher.finish()
+}
+
+pub fn symbol_name<'a, 'tcx>(instance: Instance<'tcx>, tcx: TyCtxt<'a, 'tcx, 'tcx>)
```

Comment:
Total nit, but I'd expect `tcx` to be the first argument. Having this be a method seems even better (didn't you do this elsewhere? Or is that just for `TransItem`?) But I don't care much.